pkg/cache/task.go

```diff
@@ -599,3 +601,9 @@ func (task *Task) failWithEvent(errorMessage, actionReason string) {
 	events.GetRecorder().Eventf(task.pod.DeepCopy(),
 		nil, v1.EventTypeWarning, actionReason, actionReason, errorMessage)
 }
+
+func (task *Task) setTaskPod(pod *v1.Pod) {
```

Comment:
Pardon me, why this PR does not adopt the cache solution from the discussion of https://issues.apache.org/jira/browse/YUNIKORN-2319?

Reply:
Note sure what you mean? I think you're probably referring to the suggestion of removing the pod reference from Task itself entirely and retrieving from the scheduler cache. That's possible but it's far more work. This is way simpler. Also, as long as the task object exists in memory, I think it's better to keep the pod reference.

Reply:
> I think you're probably referring to the suggestion of removing the pod reference from Task itself entirely and retrieving it from the scheduler cache. That's possible but it's far more work. This is way simpler. Also, as long as the task object exists in memory, I think it's better to keep the pod reference.

Got it and agree to this is simpler